### PR TITLE
[webnn] Add float32 tests for Element-wise logical operations

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr @Honry PTAL, thanks.

### CODE REVIEW

File: webnn/resources/utils.js [hunk 1]
```diff
@@ -277,6 +277,14 @@ const PrecisionMetrics = {
   min: {ULP: {float32: 0, float16: 0}},
   pow: {ULP: {float32: 32, float16: 2}},
   // End Element-wise binary operations
+  // Begin Element-wise logical operations
+  equal: {ULP: {uint8: 0}},
```
Comment (BruceDai) [inline webnn/resources/utils.js:281]: For logical operations, no mater input data type is float32 or float16 and etc. , the output type should be **uint8**.
And here **uint8** is align with expected data type in json file, likes

```
"expected": {
  "name": "output",
  "shape": [2, 1, 4, 1, 3],
  "data": [
    0,
    0,
    0,
    1,
    0,
    ...
    0
  ],
  "type": "uint8"
}
```

@fdwr Is it OK for this addition code? Thanks.

**Review by fdwr — APPROVED** [review]
👍

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Comment by BruceDai** [comment]
Hi @fdwr, I synced with @huningxin, we decided to rename **not** op tests to previous **logicalNot** op tests, since Chromium WebNN API implementation has already supported **logicalNot** op, then these WPT **logicalNot** tests can be ran, please take another look at this update, thanks.

**Comment by fdwr** [comment]
@BruceDai : Ok, well whatever the final outcome of the spec is, it's a simple rename either way. 👍

**Review by fdwr — APPROVED** [review]

**Comment by BruceDai** [comment]
Rebase code to check whether last two fail CIs can pass, then go to ask for merging it.

**Pull request merged**